src/oscar/apps/checkout/applicator.py

```diff
@@ -0,0 +1,81 @@
+from oscar.core import prices
+from oscar.core.loading import get_classes
+
+PercentageCharge, FlatCharge = get_classes("checkout.surcharges", ["PercentageCharge", "FlatCharge"])
+
+
+class SurchargeList(object):
```

Comment:
maybe it is better to make the SurchargeList a real list so you can loop through the methods directly:
```
class SurchargeList(list):

  @property
  def total(self):
    return sum(self).price
```
less code and no need for a constructor.